project/manager.go

```diff
@@ -58,12 +61,40 @@ func NewManager(chainEndpoint, contractAddress, projectFileDirectory string) (*M
 
 		pool[p.ID] = &p
 	}
-
-	return &Manager{
+	m := &Manager{
 		pool:            pool,
 		chainEndpoint:   chainEndpoint,
 		contractAddress: contractAddress,
-	}, nil
+	}
+
+	metas, err := m.syncProjects()
+	if err != nil {
+		return nil, errors.Wrap(err, "sync project configs from contract failed")
+	}
+
+	for _, meta := range metas {
+		if meta.Paused {
+			slog.Debug("project paused", "project_id", meta.ProjectID)
+			continue
+		}
+		c, err := meta.ProjectConfig()
+		if err != nil {
+			slog.Debug(err.Error())
+			continue
+		}
+		pool[meta.ProjectID] = &Project{
+			ID:     meta.ProjectID,
+			Config: *c,
+		}
+	}
+
+	// start monitor contract
+	go func() {
+		err := m.monitorContract()
```

Comment:
if err != nil {}

Reply:
just return err, and make NewManager return err, so main will Fatal. otherwise, monitorContract is core logic, if it failed, will not fatal or auto retry

Reply:
do you mean, if `monitorcontract` failed service should stop?